Terraform Apply of `argocd_cluster` resource for EKS is crashing the provider

```
resource "argocd_cluster" "eks" {
  for_each   = local.clusters
  server     = format("https://%s", data.aws_eks_cluster.cluster[each.value.cluster_name].endpoint)
  name       = each.value.cluster_name
  namespaces = [""]

  config {
    aws_auth_config {
      cluster_name = each.value.cluster_name
    }
    tls_client_config {
      ca_data = data.aws_eks_cluster.cluster[each.value.cluster_name].certificate_authority[0].data
    }
  }
}
```
The above code results in the following error on terraform apply (plan is successful)..

```

Stack trace from the terraform-provider-argocd_v3.2.0 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 27 [running]:
github.com/oboukili/terraform-provider-argocd/argocd.expandCluster(0xc000101e00, 0x0, 0x0, 0x0)
	github.com/oboukili/terraform-provider-argocd/argocd/structure_cluster.go:27 +0x6d0
github.com/oboukili/terraform-provider-argocd/argocd.resourceArgoCDClusterCreate(0x2c5b678, 0xc000494600, 0xc000101e00, 0x25f1f00, 0xc00013ad80, 0xc000450e00, 0x6fd38a, 0xc000bdbd60)
	github.com/oboukili/terraform-provider-argocd/argocd/resource_argocd_cluster.go:38 +0x1dd
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0008ad6c0, 0x2c5b608, 0xc0004e6880, 0xc000101e00, 0x25f1f00, 0xc00013ad80, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:330 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0008ad6c0, 0x2c5b608, 0xc0004e6880, 0xc0003e3810, 0xc000bdbd60, 0x25f1f00, 0xc00013ad80, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:456 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00069ed38, 0x2c5b608, 0xc0004e6880, 0xc000454690, 0xc0004e6880, 0x2717720, 0xc0008bdd00)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc000bd0100, 0x2c5b6b0, 0xc0004e6880, 0xc0003e3570, 0xc000bd0100, 0xc0008bdda0, 0xc00058cba0)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x2717720, 0xc000bd0100, 0x2c5b6b0, 0xc0008bdda0, 0xc0006a7d40, 0x0, 0x2c5b6b0, 0xc0008bdda0, 0xc000066a80, 0x310)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002f81c0, 0x2c75758, 0xc000024900, 0xc0003da7e0, 0xc000380b10, 0x3e3d6e0, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.40.0/server.go:1297 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0002f81c0, 0x2c75758, 0xc000024900, 0xc0003da7e0, 0x0)
	google.golang.org/grpc@v1.40.0/server.go:1626 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000396280, 0xc0002f81c0, 0x2c75758, 0xc000024900, 0xc0003da7e0)
	google.golang.org/grpc@v1.40.0/server.go:941 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.40.0/server.go:939 +0x1fd

Error: The terraform-provider-argocd_v3.2.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/argocd/resource_argocd_cluster_test.go
+++ b/argocd/resource_argocd_cluster_test.go
@@ -356,6 +356,25 @@ func TestAccArgoCDCluster_uniqueByServerTrimmed(t *testing.T) {
 	})
 }
 
+func TestAccArgoCDCluster_namespacesErrorWhenEmpty(t *testing.T) {
+	name := acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t); testAccPreCheckFeatureSupported(t, featureProjectScopedClusters) },
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccArgoCDClusterNamespacesContainsEmptyString(name),
+				ExpectError: regexp.MustCompile("namespaces: must contain non-empty strings"),
+			},
+			{
+				Config:      testAccArgoCDClusterNamespacesContainsEmptyString_MultipleItems(name),
+				ExpectError: regexp.MustCompile("namespaces: must contain non-empty strings"),
+			},
+		},
+	})
+}
+
 func testAccArgoCDClusterBearerToken(clusterName string) string {
 	return fmt.Sprintf(`
 resource "argocd_cluster" "simple" {
@@ -583,6 +602,42 @@ resource "argocd_cluster" "cluster_metadata" {
       test = "annotation"
     }
   }
+  config {
+    # Uses Kind's bootstrap token whose ttl is 24 hours after cluster bootstrap.
+    bearer_token = "abcdef.0123456789abcdef"
+    tls_client_config {
+      insecure = true
+    }
+  }
+}
+`, clusterName)
+}
+
+func testAccArgoCDClusterNamespacesContainsEmptyString(clusterName string) string {
+	return fmt.Sprintf(`
+resource "argocd_cluster" "simple" {
+  server = "https://kubernetes.default.svc.cluster.local"
+  name   = "%s"
+  shard  = "1"
+  namespaces = [""]
+  config {
+    # Uses Kind's bootstrap token whose ttl is 24 hours after cluster bootstrap.
+    bearer_token = "abcdef.0123456789abcdef"
+    tls_client_config {
+      insecure = true
+    }
+  }
+}
+`, clusterName)
+}
+
+func testAccArgoCDClusterNamespacesContainsEmptyString_MultipleItems(clusterName string) string {
+	return fmt.Sprintf(`
+resource "argocd_cluster" "simple" {
+  server = "https://kubernetes.default.svc.cluster.local"
+  name   = "%s"
+  shard  = "1"
+  namespaces = ["default", ""]
   config {
     # Uses Kind's bootstrap token whose ttl is 24 hours after cluster bootstrap.
     bearer_token = "abcdef.0123456789abcdef"

--- a/argocd/structure_cluster.go
+++ b/argocd/structure_cluster.go
@@ -1,6 +1,8 @@
 package argocd
 
 import (
+	"fmt"
+
 	application "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
@@ -24,6 +26,9 @@ func expandCluster(d *schema.ResourceData) (*application.Cluster, error) {
 	}
 	if ns, ok := d.GetOk("namespaces"); ok {
 		for _, n := range ns.([]interface{}) {
+			if n == nil {
+				return nil, fmt.Errorf("namespaces: must contain non-empty strings")
+			}
 			cluster.Namespaces = append(cluster.Namespaces, n.(string))
 		}
 	}

